### ISSUE TITLE
Add indexpack VFS for Google Cloud Storage [phab:D127]

### ISSUE THREAD

**Issue by kythe** [google/kythe#332]
_Created by **schroederc** at 2015-03-06 23:30:55_: 

Depends on #331

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-03-12 21:13:47_:

Rebase

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-04-01 15:42:26_:

Ping

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2015-04-01 23:03:22_:

Can we do something simpler here? Per http://godoc.org/path/filepath#Match

> On Windows, escaping is disabled. Instead, '\\' is treated as path separator.

which suggests the check for Windows isn't needed. (And `Glob` follows the syntax of `Match`)

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2015-04-01 23:03:22_:

You could just say `cu := i.(*apb.CompilationUnit)`. The index pack has already done this encoding (because that's the option you passed in above), so if it fails here, it //should// panic.

**Comment by kythe** [comment]
_**fromberger** wrote at 2015-04-01 23:03:22_:

Ye gods, the code for the cloud APIs is terrible. :P Your changes seem fine though. :)

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-04-02 16:13:45_:

Review changes

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-04-02 16:13:50_:

This actually was derived from that package: https://golang.org/src/path/filepath/match.go#L100

Anyway, I noticed an issue and added some tests to improve our confidence.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-04-02 16:13:50_:

Done.

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2015-04-02 16:23:47_

**Comment by kythe** [comment]
_**schroederc** closed via commit 7a02e24b8c2935934a985c5e8cca4d63473958b4 at 2015-04-02 16:29:02_
